### PR TITLE
RDCC-2390: Delete API - Delete Users by Email Pattern or User ID

### PR DESCRIPTION
**Pull request by AdilOoze**
### JIRA link (if applicable) ###

[RDCC-2390](https://tools.hmcts.net/jira/browse/RDCC-2390)

### Change description ###

New `Delete API` to Delete _(Functional Test)_ Users by `Email Pattern` or `User ID`

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by kavithagade12 — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/feign/UserProfileFeignClient.java [hunk 1]
```diff
@@ -30,11 +31,22 @@
...
+    @DeleteMapping(value = "/v1/userprofile/users?userId={userId}")
+    @RequestLine("DELETE /v1/userprofile/users?userId={userId}")
+    @Headers({"Authorization: {authorization}", "ServiceAuthorization: {serviceAuthorization}",
+        "Content-Type: application/json"})
+    Response deleteUserProfileById(@PathVariable String userId);
+
+    @DeleteMapping(value = "/v1/userprofile/users?emailPattern={emailPattern}")
+    @RequestLine("DELETE /v1/userprofile/users?emailPattern={emailPattern}")
+    @Headers({"Authorization: {authorization}", "ServiceAuthorization: {serviceAuthorization}",
+        "Content-Type: application/json"})
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/feign/UserProfileFeignClient.java:49]: 1 ) how Rest call will be differentiated here ? Delete  /v1/userprofile/users will not clash ? i think you can keep only one call here
2) These is no @PathVariable here , i think only has request param

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefUsersController.java [hunk 2]
```diff
@@ -182,4 +189,61 @@
...
+        CaseWorkerProfilesDeletionResponse resource;
+
+        if (isNotBlank(userId)) {
+            resource = caseWorkerService.deleteByUserId(userId);
+
+        } else if (isNotBlank(emailPattern)) {
+            resource = caseWorkerService.deleteByEmailPattern(emailPattern);
+
+        } else {
+            throw new InvalidRequestException("No User ID or Email Pattern provided to delete the User(s)");
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefUsersController.java:243]: I think use existing exception Forbidden with specific message. And avoid hard coded error message.

Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefUsersController.java:243]: This is in the event of a `400` `Bad Request` so I think Forbidden would not be correct here @abhijitdiwan ?

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/feign/UserProfileFeignClient.java (hunk 1 above)
Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/feign/UserProfileFeignClient.java]: Addressed here: https://github.com/hmcts/rd-caseworker-ref-api/pull/287/commits/39bd1ddb47947204d4427c142785dfb2036ade12

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefUsersController.java (hunk 2 above)
Comment (abhijitdiwan) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefUsersController.java:243]: ok ok...replace message with constant.

Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefUsersController.java]: Addressed here: https://github.com/hmcts/rd-caseworker-ref-api/pull/287/commits/4d75068b27e5a480f5dab4627536f5263326fa99

**Pull request closed without merging**